src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.cc

```diff
@@ -0,0 +1,131 @@
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "sdl_rpc_plugin/commands/hmi/on_bc_system_capability_updated_notification_from_hmi.h"
+
+#include "application_manager/display_capabilities_builder.h"
+#include "application_manager/message_helper.h"
+#include "interfaces/HMI_API.h"
+#include "interfaces/MOBILE_API.h"
+
+namespace sdl_rpc_plugin {
+using namespace application_manager;
+
+namespace commands {
+
+OnBCSystemCapabilityUpdatedNotificationFromHMI::
+    OnBCSystemCapabilityUpdatedNotificationFromHMI(
+        const application_manager::commands::MessageSharedPtr& message,
+        ApplicationManager& application_manager,
+        rpc_service::RPCService& rpc_service,
+        HMICapabilities& hmi_capabilities,
+        policy::PolicyHandlerInterface& policy_handle)
+    : NotificationFromHMI(message,
+                          application_manager,
+                          rpc_service,
+                          hmi_capabilities,
+                          policy_handle) {}
+
+OnBCSystemCapabilityUpdatedNotificationFromHMI::
+    ~OnBCSystemCapabilityUpdatedNotificationFromHMI() {}
+
+OnBCSystemCapabilityUpdatedNotificationFromHMI::
+    ProcessSystemDisplayCapabilitiesResult
+    OnBCSystemCapabilityUpdatedNotificationFromHMI::
+        ProcessSystemDisplayCapabilities(
+            const smart_objects::SmartObject& display_capabilities) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (!(*message_)[strings::msg_params].keyExists(strings::app_id)) {
+    LOG4CXX_DEBUG(logger_, "Updating general display capabilities");
+    hmi_capabilities_.set_system_display_capabilities(display_capabilities);
+    return ProcessSystemDisplayCapabilitiesResult::SUCCESS;
+  }
+
+  const auto app_id =
+      (*message_)[strings::msg_params][strings::app_id].asUInt();
+  auto app = application_manager_.application(app_id);
+  if (!app) {
+    LOG4CXX_ERROR(logger_,
+                  "Application with app_id " << app_id << " is not registered");
+    return ProcessSystemDisplayCapabilitiesResult::FAIL;
+  }
+
+  LOG4CXX_DEBUG(logger_, "Updating display capabilities for app " << app_id);
+  app->set_display_capabilities(display_capabilities);
+
+  // Remove app_id from notification to mobile
+  (*message_)[strings::params][strings::connection_key] =
+      (*message_)[strings::msg_params][strings::app_id];
+  (*message_)[strings::msg_params].erase(strings::app_id);
+  if (app->is_resuming()) {
+    LOG4CXX_DEBUG(logger_, "Application is resuming");
```

Comment:
After registering an app with the sdl hmi, this line is printing, but the app is not resumed into full. Also the RPC OnSystemCapability the hmi is sending is never being sent to mobile because of this caching logic and the failed resumption.

Reply:
I think I am experiencing and issue with resumption. app->is_resuming() is true, but resumption is not taking place successfully.

Reply:
PS. This resumption issue does not happen every time.